electrum/lnrouter.py

```diff
@@ -529,22 +529,34 @@ def get_distances(
         blacklist = self.liquidity_hints.get_blacklist()
         distance_from_start = defaultdict(lambda: float('inf'))
         distance_from_start[nodeB] = 0
-        prev_node = {}  # type: Dict[bytes, PathEdge]
+        previous_hops = {}  # type: Dict[bytes, PathEdge]
```

Comment:
why the renaming?
are you trying to reflect that the dict values are edges ("hops"?) instead of nodes?

Reply:
Yes, when first looking at this part of the code the naming was very confusing to me, especially the singular `node`, but I'm not 100% sure how to name it. What would be your interpretation of `prev_node`? Related is also the naming of the method itself.

Reply:
This dict is the analogue of the `prev` map in the [wikipedia Dijkstra pseudocode](https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm#Pseudocode). `prev[v] == u` means that the cheapest path from the start_node to `v` has the `u->v` edge as its last edge; and using the `prev` map you can construct a path from the start_node to any `v` via simple backtracking through `prev`. So in the original pseudocode the map stores the previous node on the cheapest path to a given node. In our code the value is the `u->v` edge instead of the `u` node.

Reply:
> also the word prev could be confusing.

Hmm, what other meaning do you see? :)
Maybe I have tunnel vision here; especially as it was probably I who named this variable...

Reply:
Ok, thanks for the explainer. I think the confusion just came from the directionality in connection with the word previous, as we search from the end node and go back to the start node. But it's difficult to explain why oneselve's confused :).

Reply:
well ok, I don't mind either naming; so we can rename it if you think it's clearer now